tidy3d/components/data/data_array.py

```diff
@@ -551,6 +551,42 @@ class ChargeDataArray(DataArray):
     _dims = ("n", "p")
 
 
+class PointDataArray(DataArray):
+    """Indexed data array.
+
+    Example
+    -------
+    >>> td = IndexedDataArray((1+1j) * np.random.random((3,)), coords=dict(index=np.arange(3)))
+    """
+
+    __slots__ = ()
+    _dims = ("index", "axis")
+
+
+class CellDataArray(DataArray):
```

Comment:
can you explain this a bit? so `CellDataArray[i,j]` is the `j`-th vertex of cell `i`? And what does it store? an integer  index into another cell? or into the point data array?  When I worked on unstructured meshes I would usually have a 3D array with dims `cell_index, vertex_index, axis`. It seems that maybe this kind of data structure could be constructed out of this cell data array + the point data array?

Reply:
Yes, exactly, it stores indices of points that constitute the cell. And the representation that you mentioned can definitely be constructed from `PointDataArray` + `CellDataArray`